### ISSUE TITLE
cpu count versus available cpus

### ISSUE THREAD

**Issue by EricDeveaud** [oschwengers/bakta#135]
Hello,

bakta uses multiprocessing.cpu_count() to ge the number of cpus in `bakta/config.py`, `bakta/proteins.py` and `bakta/utils.py`.
`multiprocessing.cpu_count()` returns the number of cpu in the machine, But this is not the same as the number of cpu available to the process. For example, you can run in a taskset context or a batch scheduler like slurm.

see:
```
$ nproc
96
$ taskset -c 1 nproc
1
$ taskset -c 1 python3 -c "import multiprocessing; print(multiprocessing.cpu_count())"
96
```

I would suggest to use `len(os.sched_getaffinity(0))` instead of `multiprocessing.cpu_count()`

```
$ python3 -c "import os; print(len(os.sched_getaffinity(0)))"
96
$ taskset -c 1 python3 -c "import os; print(len(os.sched_getaffinity(0)))"
1
```
NB Mac OSX python does not have os.sched_getaffinity so a portable way to code it would be eg:

```
try:
    num_cpus = len(os.sched_getaffinity(0))
except AttributeError:
    num_cpus = multiprocessing.cpu_count()
```

regards

Eric

**Comment by oschwengers** [comment]
Hi @EricDeveaud,
thanks for the heads-up and code examples. Can't you just use the `--threads` parameter to reduce the number of used threads? When submitting Bakta jobs to SLURM, wouldn't you set a specific number of CPU slots via `-c <CPUS>`? Then you could also tell Bakta to just use exaclty that amount of threads via `--threads <CPUS>`.
I'm just asking to get a better understanding. I think it shouldn't be a problem to add your suggestions.

Best regards!
Oliver

**Comment by oschwengers** [comment]
Thanks again!
A patch is now merged. It will be available with the upcoming patch release [v1.5.1](https://github.com/oschwengers/bakta/milestone/9)
